csharp/src/Google.Protobuf.Test/ExtensionSetTest.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Google.Protobuf.TestProtos.Proto2;
+using NUnit.Framework;
+
+using static Google.Protobuf.TestProtos.Proto2.UnittestExtensions;
+
+namespace Google.Protobuf
+{
+    public class ExtensionSetTest
+    {
+        [Test]
+        public void EmptyExtensionSet()
+        {
+            ExtensionSet<TestAllExtensions> extensions = new ExtensionSet<TestAllExtensions>();
+            Assert.AreEqual(0, extensions.CalculateSize());
+        }
+
+        [Test]
+        public void MergeExtensionSet()
+        {
+            ExtensionSet<TestAllExtensions> extensions = null;
+            ExtensionSet.Set(ref extensions, OptionalBoolExtension, true);

```

Comment:
Is it possible to change the API not static and the usage to extensions.Set(OptionalBoolExtension, true)?  Also for other methods like Get/Clear/Has... 
Sorry I didn't point this out in the extension PR

Reply:
We could, however this API allows us to have greater control whether an extension set is set in a field or not and isn't exposed as part of messages so I don't believe it matters too much? 

The tests show that you start out without an extension set (so a message doesn't create an extension set as part of construction which saves space), and by setting a value, a set is created automatically and assigned to the field. That way we don't have to null-check ourselves in each message and it can be done as part of the set operation in each ExtensionSet helper method.

Reply:
Make sense to me